### DISCUSSION TITLE
Is there a competition between libraries for I2S?

### DISCUSSION THREAD

**Discussion by renatoianhez** [pschatzmann/ESP32-A2DP#547] [Q&A] [answered]
With the new version of the ESP32-Arduino IDE, there was a change in the I2S API, and consequently some libraries were affected. One of them that I use is ESP32-A2DP, as we saw. On [this version of my webradio](https://github.com/renatoianhez/Lolin32-radio/tree/main/RadioWebTestesESP3) I use it to transform it into a bluetooth receiver speaker. However, updating that library seems to conflict with the [ESP32-audioI2S library](https://github.com/schreibfaul1/ESP32-audioI2S), since, even in exclusive programming blocks, the Bluetooth library does not work. However, the A2DP works in an [example](https://github.com/pschatzmann/ESP32-A2DP/tree/main/examples/bt_music_receiver_simple_with_pin) sketch. Is there a "competition" for I2S in this case? Could anyone help me with an example of integrating the two libraries?
Thanks!

**Comment by pschatzmann** [comment, marked as answer]
From a design point of view it would be better if you just use 1 library to do the output: This way you can avoid that you need to program the setup twice and you can avoid any conflicts.

In your case this is unfortunately not possible with your selected library, so you would need to make sure that the I2s is properly closed in one library before you start the output with the other library. 

I am not aware of any other potential conflicts and you would need to build some test cases to identify the root cause...

**Comment by Marco1971Repo** [comment]
I did exactly what you need and it work.

[https://github.com/Father-Son/ESP32_WebRadio](url)

@renatoianhez you can see my project

**Reply by renatoianhez** [reply]
I saw that you used the same as [Wolle's suggestion](https://github.com/schreibfaul1/ESP32-audioI2S/discussions/734)... But 'I2SStream' was not recognized... I used 'I2SClass' but it didn't work. Where can I find the AudioBoard library?

**Reply by Marco1971Repo** [reply]
To use I2SStreamyou need https://github.com/pschatzmann/arduino-audio-tools.git library as per suggestion of Phil https://github.com/pschatzmann/ESP32-A2DP?tab=readme-ov-file
